### PR TITLE
added zero-to-hero docs for twitter dapp

### PR DESCRIPTION
**Pull request by tusharojha**
@olehmell @Yung-Beef please review. 

Sharing a video too.

### CODE REVIEW

File: docs/develop/zero-to-hero/twitter-dapp/CreatingNewProject.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+id: creating-project
+title: Creating New Project
+displayed_sidebar: developSidebar
+---
+
+Let's use [Subsocial Starter](/docs/develop/developer-quickstart) to create a new project named **twitter-dapp**
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/CreatingNewProject.md:7]:
```suggestion
Let's use the [Subsocial Starter](/docs/develop/developer-quickstart) to create a new project named **twitter-dapp**
```

File: docs/develop/zero-to-hero/twitter-dapp/CreatingNewProject.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+:::info Note
+Make sure you have [node](https://nodejs.org), npm and [yarn](https://classic.yarnpkg.com/lang/en/docs/install/#mac-stable) installed before going forward.
+:::
+
+### Follow the steps 
+
+1. Open terminal inside a directory of your choice.
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/CreatingNewProject.md:15]:
```suggestion
1. Open the terminal inside a directory of your choice.
```

File: docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+id: finish-app
+title: Finishing Dapp
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md:3]:
```suggestion
title: Finishing The Dapp
```

File: docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+displayed_sidebar: developSidebar
+---
+
+## Congratulations 🥂🎉
+
+Now, you have your very own twitter-like Dapp on Subsocial.
+
+Share it with people to show-off your skills and learning.
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md:11]:
```suggestion
Share it with people to show off your skills and learning.
```

Comment (Yung-Beef) [reply docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md:11]: and maybe just put the below hyperlink onto "Share" in this line and delete line 13

File: docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+[Share on Twitter](https://ctt.ac/6G7EI)
+
+### What's next? 🤔
+
+**Subsocial** helps you build the next generation social media in no time. Here are a couple of things to do next:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md:17]:
```suggestion
**Subsocial** helps you build the next generation of social media in no time. Here are a couple of things to do next:
```

File: docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+- Experiment more with our SDK using the [Playground](https://play.subsocial.network)
+- Check the hackathons & bounties to build next, [here](/docs/bounties).
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md:20]:
```suggestion
- Check out the [hackathons & bounties](/docs/bounties).
```

File: docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+- Connect with us on [Telegram Group](https://t.me/+ZzvLu0ZfkQwxNGQy) and share your experience.
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md:21]:
```suggestion
- Connect with us on [Telegram](https://t.me/+ZzvLu0ZfkQwxNGQy) and share your experience.
```

File: docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+> Learn more about different components that make up Subsocial structures called [Entities](/docs/develop/concepts/entities).
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/FinishingApp.md:23]:
```suggestion
> Learn more about the different components that make up Subsocial's structures called [Entities](/docs/develop/concepts/entities).
```

File: docs/develop/zero-to-hero/twitter-dapp/HomeScreen.md [hunk 9]
```diff
@@ -0,0 +1,239 @@
+---
+id: home-screen-ui
+title: Home Screen UI
+displayed_sidebar: developSidebar
+---
+
+Let's design the UI for homescreen with tweets list.
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/HomeScreen.md:7]: Let's design the UI for the homescreen with a tweets list.
Let's design the UI for the homescreen with a list of tweets.

#2 sounds better to me, but up to you which one seems more correct, as I'm not 100% on the context here.

File: docs/develop/zero-to-hero/twitter-dapp/HomeScreen.md [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+
+```sh
+  yarn add react-list react-contenteditable
+```
+
+#### React Components
+
+Follow the steps to add the UI:
+
+1. Create a new file inside **src/components** directory, with name **tweet-box.tsx**. And add the following code for tweet box:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/HomeScreen.md:22]:
```suggestion
1. Create a new file inside the **src/components** directory, with the name **tweet-box.tsx**, and add the following code for tweet box:
```

File: docs/develop/zero-to-hero/twitter-dapp/HomeScreen.md [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+          </div>
+          <div className="h-2 bg-gray-100 w-full bg-opacity-15"></div>
+        </div>
+      );
+    };
+
+    export default TweetBox;
+  ```
+
+2. Now, let's create the UI of a tweet. Create a new file inside **src/components** directory, with name **tweet.tsx**. Paste the following code for UI:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/HomeScreen.md:112]:
```suggestion
2. Now, let's create the UI of a tweet. Create a new file inside the **src/components** directory, with the name **tweet.tsx**, and paste the following code for the UI:
```

File: docs/develop/zero-to-hero/twitter-dapp/HomeScreen.md [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+        </div>
+      );
+    };
+
+    export default TweetComponent;
+
+  ```
+
+
+3. Create a new file inside **src/screens** directory, with name **home.tsx**. Paste the following code for UI:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/HomeScreen.md:202]:
```suggestion
3. Create a new file inside the **src/screens** directory, with the name **home.tsx**, and paste the following code for the UI:
```

File: docs/develop/zero-to-hero/twitter-dapp/LikeaTweet.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
+---
+id: like-tweet
+title: Liking a Tweet
+displayed_sidebar: developSidebar
+---
+
+We love interacting with tweets by giving them a ❤️
+
+Let's add this functionality on the Dapp, here's an example:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LikeaTweet.md:9]:
```suggestion
Let's add this functionality on the Dapp. Here's an example:
```

File: docs/develop/zero-to-hero/twitter-dapp/LikeaTweet.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+```tsx
+  const postId = '1' // Post Id you want to react on.
+  const substrateApi = await api.blockchain.api
+
+  const reactionTx = substrateApi.tx.reactions.createPostReaction(postId, 'Upvote')
+```
+
+It's time to add this in the codebase inside the **tweet.tsx** like this:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LikeaTweet.md:18]:
```suggestion
It's time to add this in the codebase inside of **tweet.tsx** like this:
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 15]
```diff
@@ -0,0 +1,119 @@
+---
+id: login-screen
+title: Login Screen
+displayed_sidebar: developSidebar
+---
+
+Now, let's first write the login / signup screen. 
+
+In web3, user authentication is done via wallets. In this case, we will be using [Polkadotjs](https://polkadot.js.org/extension/).
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:9]: Can we do Talisman instead?

Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:9]:
```suggestion
In web3, user authentication is done via wallets. In this case, we will be using the [Polkadot.js](https://polkadot.js.org/extension/) wallet extension.
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+
+This how it looks:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:11]:
```suggestion
This is how it looks: 
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+
+This how it looks: 
+
+![Login Screen](../../../../static/img/twitter-dapp/login.png)
+
+### Creating the login screen UI
+
+Follow the steps to create this login screen: 
+
+1. Create a new file with name `login.tsx` inside a new directory **screens** in `src` folder.
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:19]:
```suggestion
1. Create a new file with the name `login.tsx` inside a new directory **screens** in the `src` folder.
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+        </button>
+      </div>
+    </div>
+  );
+}
+
+export default LoginScreen
+```
+
+3. Download the `svgs`, and save the directory inside `src` folder (if not already). [Link Here](https://drive.google.com/drive/folders/1l6t06P1aJqTCM7vi5rU6qIhwxeHc3_WR?usp=sharing)
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:56]:
```suggestion
3. [Download](https://drive.google.com/drive/folders/1l6t06P1aJqTCM7vi5rU6qIhwxeHc3_WR?usp=sharing) the `svgs` and save the directory inside the `src` folder (if it's not there already).
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+
+
+### Connecting to wallet
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:59]:
```suggestion
### Connecting to the wallet
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 20]
```diff
@@ -0,0 +1,119 @@
...
+
+For connecting the web app to wallet, we need to do 2 things:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:61]:
```suggestion
For connecting the web app to the wallet, we need to do 2 things:
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 21]
```diff
@@ -0,0 +1,119 @@
...
+
+- Use the methods provided by Subsocial Starter to connect to wallet
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:63]:
```suggestion
- Use the methods provided by the Subsocial Starter to connect to the wallet
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 22]
```diff
@@ -0,0 +1,119 @@
...
+- Saving / Fetching selected wallet address to localstorage
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:64]:
```suggestion
- Saving / Fetching selected the wallet address to localstorage
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 23]
```diff
@@ -0,0 +1,119 @@
...
+
+
+#### Fetching list of all accounts
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:67]:
```suggestion
#### Fetching a list of all accounts
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 24]
```diff
@@ -0,0 +1,119 @@
...
+
+Update the **App.tsx** file above return statement like this, and clear everything else:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:69]:
```suggestion
Update the **App.tsx** file above the return statement like this, and clear everything else:
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md [hunk 25]
```diff
@@ -0,0 +1,119 @@
...
+    } else {
+      alert('No accounts found in polkadotjs wallet.')
+    }
+  }
+
+  return ...
+}
+```
+
+#### Rendering UI conditionally
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:95]:
```suggestion
#### Rendering the UI conditionally
```

File: docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md [hunk 26]
```diff
@@ -0,0 +1,36 @@
+---
+id: introduction
+title: Introduction
+displayed_sidebar: developSidebar
+---
+
+**Zero to Hero tutorial** is your chance to build a fully decentralized twitter app under minutes using the Subsocial Starter.
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md:7]:
```suggestion
The **Zero to Hero tutorial** is your chance to build a fully decentralized twitter app in minutes using the Subsocial Starter.
```

Comment (Yung-Beef) [reply docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md:7]: or "in under __ minutes"
30 minutes? 60 minutes?

File: docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md [hunk 27]
```diff
@@ -0,0 +1,36 @@
...
+
+:::info Tip
+It's recommended to follow the steps as you read, tweet about your journey to invite people to it.
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md:10]:
```suggestion
It's recommended to follow the steps as you read and tweet about your journey to invite others as well.
```

File: docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md [hunk 28]
```diff
@@ -0,0 +1,36 @@
...
+:::
+
+### So what are we going to build? 🧐
+
+A decentralized twitter that makes your users own the content they create, audience they engage, manage follow / followings and go live instantly :D
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md:15]:
```suggestion
A decentralized twitter that lets your users own their social connections and content they create :D
```

File: docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+
+#### Sign-in using [Polkadot.js wallet](https://polkadot.js.org/extension/)
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md:17]:
```suggestion
#### Sign-in using the [Polkadot.js wallet](https://polkadot.js.org/extension/)
```

File: docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md [hunk 30]
```diff
@@ -0,0 +1,36 @@
...
+
+#### See feed / posts of a space 
+
+
+#### Tweet, Like, View Profiles
+
+
+### In case you face some issues 🐞
+
+If you face any issues, doubts or have any questions for us. Feel free to chat with us:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md:28]:
```suggestion
If you face any issues, doubts, or have any questions for us, feel free to chat with us on [Discord](https://discord.gg/WPdjvs5uWJ) and [Telegram](https://t.me/+ZzvLu0ZfkQwxNGQy).
```

File: docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md [hunk 31]
```diff
@@ -0,0 +1,36 @@
...
+
+- Discord Channel: [#developers](https://discord.gg/WPdjvs5uWJ)
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md:30]: delete

File: docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+- Telegram Channel: [Subsocial Devs](https://t.me/+ZzvLu0ZfkQwxNGQy)
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/ProjectSetup.md:31]: delete

File: docs/develop/zero-to-hero/twitter-dapp/Styling.md [hunk 33]
```diff
@@ -0,0 +1,105 @@
...
+          lg: "0.5rem",
+          full: "9999px",
+          xl: "1rem",
+        },
+      },
+      ...
+    };
+  ```
+
+4. Create a new file named **index.css** inside **src** directory, and paste the following code:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/Styling.md:74]:
```suggestion
4. Create a new file named **index.css** inside the **src** directory, and paste the following code: 
```

File: docs/develop/zero-to-hero/twitter-dapp/Styling.md [hunk 34]
```diff
@@ -0,0 +1,105 @@
...
+    @tailwind base;
+    @tailwind components;
+    @tailwind utilities;
+
+    body {
+      background-color: rgb(21, 32, 43);
+    }
+  ```
+
+5. Last step, update the following scripts in package.json:
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/Styling.md:86]:
```suggestion
5. Update the following scripts in package.json:
```

File: docs/develop/zero-to-hero/twitter-dapp/Styling.md [hunk 35]
```diff
@@ -0,0 +1,105 @@
...
+    } 
+  ```
+
+6. Add the import statement inside the **index.ts** file like this:
+
+  ```ts
+    import './assets/main.css';
+  ```
+
+7. Download the `svgs`, and save the directory inside `src` folder. [Link Here](https://drive.google.com/drive/folders/1l6t06P1aJqTCM7vi5rU6qIhwxeHc3_WR?usp=sharing)
```
Comment (Yung-Beef) [inline docs/develop/zero-to-hero/twitter-dapp/Styling.md:105]:
```suggestion
7. [Download](https://drive.google.com/drive/folders/1l6t06P1aJqTCM7vi5rU6qIhwxeHc3_WR?usp=sharing) the `svgs`, and save the directory inside `src` folder.
```

File: docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md (hunk 15 above)
Comment (tusharojha) [reply docs/develop/zero-to-hero/twitter-dapp/LoginScreen.md:9]: We need to add it to starter yet, our starter supports only polkadotjs only for now.

**Pull request merged**